src/QirRuntime/test/Standalone/InputTypes/qir-driver.cpp

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <iostream>
+#include "../Shared/CLI11.hpp"
+
+// Can manually add calls to DebugLog in the ll files for debugging.
+extern "C" void DebugLog(int64_t value)
+{
+    std::cout << value << std::endl;
+}
+extern "C" void DebugLogPtr(char* value)
+{
+    std::cout << (const void*)value << std::endl;
+}
+
+extern "C" void SetupQirToRunOnFullStateSimulator();
```

Comment:
Please don't use this factory. It's not exposed in header files for a reason :) Instead, include context.hpp, QuantumApi_I.hpp, and SimFactory.hpp and use the following pattern in the code:
```
    std::unique_ptr<ISimulator> sim = CreateFullstateSimulator();
    QirContextScope qirctx(sim.get(), false /*trackAllocatedObjects*/);
```

Reply:
I'll change it to the pattern you are suggesting.

Reply:
Not using the not-exposed factory anymore.